sysl2/grammarParser.go

```diff
@@ -184,7 +182,7 @@ func buildFirstFollowSet(g *sysl.Grammar) (map[string]*Set, map[string]*Set) {
                 for i := range seq.Term {
                     // follow
                     l := len(seq.Term)
-                    if (i + 1) < l {
+                    if i+1 < l {
```

Comment:
`if i + 1 < l … else if i < l …` is a little confusing. The equivalent `if i < l - 1 … else if i == l - 1` is easier to grok. Additionally, you could use `last := len(seq.Term) - 1` instead of `l := len(seq.Term)`.

Reply:
Agreed on len() check. Moved it out of the loop.
The idea of `i + 1`  is to check if we can access the next element too. To me this seems more natural. But not a big deal.